WordPress/Classes/ViewRelated/Reader/ReaderPostsViewController.m

```diff
@@ -870,8 +840,13 @@ - (void)cleanupAfterRefresh
 
 - (void)tableViewHandlerWillRefreshTableViewPreservingOffset:(WPTableViewHandler *)tableViewHandler
 {
-    [UIView performWithoutAnimation:^{
-        [self cleanupAfterRefresh];
+    [[self managedObjectContext] performBlockAndWait:^{
```

Comment:
Are we 1000000% sure we wanna use `performBlockAndWait` here?.
I usually try to stay away from that one... since it may cause deadlocks, and it will definitely freeze the main thread.

I've seen deadlocks of a derived context waiting to merge back into the main context, while the main thread was frozen, waiting on a derived context (related to the way the Simperium's writer / derived MOC work)